api/routes/analysis.py

```diff
@@ -309,3 +320,67 @@ async def get_sample_reads_map_for_seqr(
         media_type='text/tab-separated-values',
         headers={'Content-Disposition': f'filename={basefn}.tsv'},
     )
+
+
+@router.post('/sample-file-sizes', operation_id='getSampleFileSizes')
+async def get_sample_file_sizes(
+    start_date: str = None,
+    end_date: str = None,
+    project_names: List[str] = None,
+    connection: Connection = get_projectless_db_connection,
+) -> list[ProjectSizeModel]:
+    """
+    Get the per sample file size by type over the given projects and date range
+    """
+    atable = AnalysisLayer(connection)
+    stable = SampleLayer(connection)
+
+    # Check access to projects
+    project_ids = None
+    if project_names:
```

Comment:
Should project_names a required field, and can we raise a 400 if no projects are specified?

Reply:
Sure thing!